### DISCUSSION TITLE
Support creation of `mediumtext` and `longtext` field types

### DISCUSSION THREAD

**Discussion by t7tran** [directus/directus#6761] [Feature Requests]
At the moment, when adding a new text field, only `String` and `Text` are available with respective fixed max length values. However, if a column of type `mediumtext` or `longtext` exists in the database, it is always infered as `Text` in the UI with the correct `Length`.

The question is: should we be able to create those two field types via the UI or api? Is this an unexpected bug?

**Comment by rijkvanzanten** [comment]
> The question is: should we be able to create those two field types via the UI or api? Is this an unexpected bug?

No, as it's not something that works cross database vendor. We could expose a "use custom type" functionality that would allow you to enter any SQL specific type, but that would still have to be abstracted to a type that Directus can work with for interfaces etc

**Reply by t7tran** [reply]
Agreed. That's why I wasn't sure if it was a bug. But it seems logical to me to be able to create such a field via admin UI as it interprets existing fields. Otherwise, one would have to add (`hacking`) a field via SQL before decorating it in the app.

**Reply by t7tran** [reply]
So any support via UI would be greatly appreciated.

**Reply by rijkvanzanten** [reply]
We could have an "allow other" on that type select dropdown, and then have a check in the API to see if it's one of the knex built-in types, or a custom type to auto-switch between calling the method on knex or using `table.specificType`

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
